Project lint fails due to expired TODO comment in NameConfiguration.swift

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The deadline of the following TODO comment has been reached and causes lint error.
https://github.com/realm/SwiftLint/blob/ad23d08cdaa7d656a6e96d6f6cc7b0d165f0a8f2/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/NameConfiguration.swift#L77

Cloning this project and running `swiftlint lint .` will reproduce this issue.

```sh
/Users/username/Projects/SwiftLint/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/NameConfiguration.swift:77:23: error: Expiring Todo Violation: TODO/FIXME has expired and must be resolved (expiring_todo)
Done linting! Found 1 violation, 1 serious in 640 files.
```

![Image](https://github.com/user-attachments/assets/624a7718-3b65-45be-a2f8-ad905a344fcf)

### Environment

* SwiftLint version: 0.59.1
* Xcode version: Xcode 16.1
* Installation method used: homebrew

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,8 +4,8 @@
 
 ### Breaking
 
-* Removed support for boolean values in the `validates_start_with_lowercase` option
-  and its deprecation warning. Use severity levels (`off`, `warning`, or `error`) instead.
+* Remove support for boolean values in the `validates_start_with_lowercase` option entirely.
+  Use the severity levels `off`, `warning` or `error` instead.  
   [kaseken](https://github.com/kaseken)
 
 ### Experimental

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,9 @@
 
 ### Breaking
 
-* None.
+* Removed support for boolean values in the `validates_start_with_lowercase` option
+  and its deprecation warning. Use severity levels (`off`, `warning`, or `error`) instead.
+  [kaseken](https://github.com/kaseken)
 
 ### Experimental
 

--- a/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/NameConfiguration.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/NameConfiguration.swift
@@ -73,16 +73,6 @@ struct NameConfiguration<Parent: Rule>: RuleConfiguration, InlinableOptionType {
         }
         if let validatesStartWithLowercase = configurationDict[$validatesStartWithLowercase.key] as? String {
             try self.validatesStartWithLowercase.apply(configuration: validatesStartWithLowercase)
-        } else if let validatesStartWithLowercase = configurationDict[$validatesStartWithLowercase.key] as? Bool {
-            // TODO: [05/10/2025] Remove deprecation warning after ~2 years.
-            self.validatesStartWithLowercase = validatesStartWithLowercase ? .error : .off
-            Issue.genericWarning(
-                """
-                The \"validates_start_with_lowercase\" configuration now expects a severity (warning or \
-                error). The boolean value 'true' will still enable it as an error. It is now deprecated and will be \
-                removed in a future release.
-                """
-            ).print()
         }
     }
 }

--- a/Source/SwiftLintBuiltInRules/Rules/Style/IdentifierNameRuleExamples.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Style/IdentifierNameRuleExamples.swift
@@ -33,11 +33,6 @@ internal struct IdentifierNameRuleExamples {
     ]
 
     static let triggeringExamples = [
-        Example(
-            "let ↓MyLet = 0",
-            configuration: ["validates_start_with_lowercase": true],
-            excludeFromDocumentation: true
-        ),
         Example("class C { static let ↓_myLet = 0 }"),
         Example("class C { class let ↓MyLet = 0 }"),
         Example("class C { static func ↓MyFunc() {} }"),

--- a/Tests/BuiltInRulesTests/GenericTypeNameRuleTests.swift
+++ b/Tests/BuiltInRulesTests/GenericTypeNameRuleTests.swift
@@ -58,6 +58,6 @@ final class GenericTypeNameRuleTests: SwiftLintTestCase {
 
         let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
                                          .with(triggeringExamples: triggeringExamples)
-        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": false])
+        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": "off"])
     }
 }

--- a/Tests/BuiltInRulesTests/IdentifierNameRuleTests.swift
+++ b/Tests/BuiltInRulesTests/IdentifierNameRuleTests.swift
@@ -62,7 +62,7 @@ final class IdentifierNameRuleTests: SwiftLintTestCase {
         let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
                                          .with(triggeringExamples: triggeringExamples)
 
-        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": false])
+        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": "off"])
     }
 
     func testStartsWithLowercaseCheck() {
@@ -81,14 +81,14 @@ final class IdentifierNameRuleTests: SwiftLintTestCase {
             IdentifierNameRule.description
                 .with(triggeringExamples: triggeringExamples)
                 .with(nonTriggeringExamples: nonTriggeringExamples),
-            ruleConfiguration: ["validates_start_with_lowercase": true]
+            ruleConfiguration: ["validates_start_with_lowercase": "error"]
         )
 
         verifyRule(
             IdentifierNameRule.description
                 .with(triggeringExamples: [])
                 .with(nonTriggeringExamples: nonTriggeringExamples + triggeringExamples.removingViolationMarkers()),
-            ruleConfiguration: ["validates_start_with_lowercase": false]
+            ruleConfiguration: ["validates_start_with_lowercase": "off"]
         )
     }
 
@@ -103,7 +103,7 @@ final class IdentifierNameRuleTests: SwiftLintTestCase {
                     Example("enum Foo { case myCase }"),
                 ]),
             ruleConfiguration: [
-                "validates_start_with_lowercase": true,
+                "validates_start_with_lowercase": "error",
                 "allowed_symbols": ["M"],
             ] as [String: any Sendable]
         )

--- a/Tests/BuiltInRulesTests/NameConfigurationTests.swift
+++ b/Tests/BuiltInRulesTests/NameConfigurationTests.swift
@@ -47,21 +47,6 @@ final class NameConfigurationTests: SwiftLintTestCase {
         XCTAssertEqual(nameConfig.validatesStartWithLowercase, .warning)
     }
 
-    func testNameConfigurationWithDeprecatedBooleanSeverity() throws {
-        var nameConfig = TesteeType(minLengthWarning: 0,
-                                    minLengthError: 0,
-                                    maxLengthWarning: 0,
-                                    maxLengthError: 0)
-
-        XCTAssertEqual(nameConfig.validatesStartWithLowercase, .error)
-
-        try nameConfig.apply(configuration: ["validates_start_with_lowercase": false])
-        XCTAssertEqual(nameConfig.validatesStartWithLowercase, .off)
-
-        try nameConfig.apply(configuration: ["validates_start_with_lowercase": true])
-        XCTAssertEqual(nameConfig.validatesStartWithLowercase, .error)
-    }
-
     func testNameConfigurationThrowsOnBadConfig() {
         let config = 17
         var nameConfig = TesteeType(minLengthWarning: 0,

--- a/Tests/BuiltInRulesTests/TypeNameRuleTests.swift
+++ b/Tests/BuiltInRulesTests/TypeNameRuleTests.swift
@@ -58,6 +58,6 @@ final class TypeNameRuleTests: SwiftLintTestCase {
         let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
                                          .with(triggeringExamples: triggeringExamples)
 
-        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": false])
+        verifyRule(description, ruleConfiguration: ["validates_start_with_lowercase": "off"])
     }
 }

